### PR TITLE
Add BYOK feature support to Terraform

### PR DESCRIPTION
**Pull request by arishta-yb**
Summary:

This PR adds encryption at rest support to Terraform.
1. Cluster creation with AWS CMK and GCP CMK 
2. Adding EAR to existing cluster 
3. Enable or disable EAR for existing EAR enabled cluster.

Testing:

Manually tested out all the scenarios.

### CODE REVIEW

File: managed/resource_cluster.go
Comment (posriniv) [inline managed/resource_cluster.go:518]: Maybe check `!= aws`, any other thing string should not be allowed. You can also consider using a validator like this: https://github.com/yugabyte/terraform-provider-ybm/blob/main/managed/resource_cluster.go#LL67C5-L67C89

**Review by posriniv — CHANGES_REQUESTED** [review]
Please check the update cluster flow also.

**Comment by arishta-yb** [comment]
> Please check the update cluster flow also.

Will take the update cluster flow in the next PR.

File: managed/resource_cluster.go
Comment (posriniv) [inline managed/resource_cluster.go:1165]: Can common code be extracted out of this switch case and the switch case statement above?

**Review by posriniv — COMMENTED** [review]
Minor comments, let's wait for a review from @cdavid too.

File: managed/resource_cluster.go
Comment (arishta-yb) [reply managed/resource_cluster.go:286]: Was not an intended change, will revert.

File: managed/resource_cluster.go
Comment (arishta-yb) [reply managed/resource_cluster.go:1165]: The above switch case is building a openapiclient.CMKSpec, however this is building the go model CMKSpec.
Can you please tell how to common out the code in both the cases?

**Review by posriniv — APPROVED** [review]
Looks good to me!

**Comment by cdavid** [comment]
Can you please add an example for each of the types of CMK clusters under: `https://github.com/yugabyte/terraform-provider-ybm/tree/main/examples/resources/ybm_cluster` ?

File: managed/models.go
Comment (cdavid) [inline managed/models.go:39]: Isn't this implicit if this structure `!= nil`? Can you specify an EAR, but have `is_enabled = false`?

File: docs/resources/cluster.md
Comment (cdavid) [inline docs/resources/cluster.md:316]: @shreyas-agnihotri @NValsan  - is CMK the external name of the feature? Do we need a better string here?

File: managed/resource_cluster.go
Comment (cdavid) [inline managed/resource_cluster.go:609]: Instead of a `(bool, string)`, look into the other places where we return an `Error` object. That is the `go way`.

File: managed/resource_cluster.go
Comment (cdavid) [inline managed/resource_cluster.go:1085]: Did you try applying the same template multiple times? I believe you are not populating `is_enabled`, so it might break...

File: docs/resources/cluster.md
Comment (shreyas-agnihotri) [reply docs/resources/cluster.md:316]: Yeah, we are calling it CMK in docs. https://docs.yugabyte.com/preview/yugabyte-cloud/cloud-secure-clusters/managed-ear/

File: managed/resource_cluster.go
Comment (arishta-yb) [reply managed/resource_cluster.go:1085]: Yes, it works when I apply the same template multiple times. 
`is_enabled` is populated here inside the `resourceClusterRead` function. 

https://github.com/yugabyte/terraform-provider-ybm/blob/86be15a353ea7f62feee3944ed4fa4a1492d0147/managed/resource_cluster.go#L1163

The code being referred here assigns the credentials field (secret key/access key, client id/private key) to their original value (since the GET API returns a masked value).

File: managed/models.go
Comment (arishta-yb) [reply managed/models.go:39]: Yes, it is supported to create an EAR for existing cluster with `is_enabled = false`.

File: managed/resource_cluster.go
Comment (arishta-yb) [reply managed/resource_cluster.go]: Done

**Review by cdavid — APPROVED** [review]

**Pull request merged**
